packages/core/src/components/DataTable/index.tsx

```diff
@@ -0,0 +1,437 @@
+        selectedChildren.delete(originalIndex);
+        // If there are now 0 selected children
+        if (selectedChildren.size === 0) {
+          delete selectedRows[parentOriginalIndex!];
+          // If there is still at least one selected child
+        } else {
+          selectedRows[parentOriginalIndex!].status = SELECTION_OPTIONS.HAS_ACTIVE_CHILD;
+        }
+        // If child is not already selected
+      } else {
+        selectedChildren.add(originalIndex);
+        // If all children are now selected
+        if (sortedDataList[parentIndex!].metadata.children.length === selectedChildren.size) {
+          selectedRows[parentOriginalIndex!].status = SELECTION_OPTIONS.ACTIVE;
+          // If not all children are now selected
+        } else {
+          selectedRows[parentOriginalIndex!].status = SELECTION_OPTIONS.HAS_ACTIVE_CHILD;
+        }
+      }
+      // If parent is not already selected
+    } else {
+      selectedRows[parentOriginalIndex!] = {
+        status: SELECTION_OPTIONS.HAS_ACTIVE_CHILD,
+        selectedChildren: new Set([originalIndex]),
+      };
+    }
+
+    this.setState({
+      sortedDataList,
+    });
+  };
+
+  private handleParentSelection(row: ExpandedRow) {
+    const { selectedRows }: { selectedRows: SelectedRows } = this.state;
+    const { originalIndex } = row.metadata;
+
+    // If parent is already selected
+    if (Object.prototype.hasOwnProperty.call(selectedRows, originalIndex)) {
+      delete selectedRows[originalIndex];
+      // If parent is not already selected
+    } else {
+      // Is there are children, select them all
+      const children = row.metadata.children
+        ? row.metadata.children.map((child: IndexedChildRow) => child.metadata.originalIndex)
+        : [];
+      selectedRows[originalIndex] = {
+        status: SELECTION_OPTIONS.ACTIVE,
+        selectedChildren: new Set(children),
+      };
+    }
+
+    this.setState({ selectedRows });
+  }
+
+  private handleSelection = (rowData: ExpandedRow) => () => {
+    if (rowData.metadata.isChild) {
+      this.handleChildSelection(rowData);
+    } else {
+      this.handleParentSelection(rowData);
+    }
+  };
+
+  // Have to use any to match react-virutalized's specified callback signature
+  private handleRowClick = ({ rowData }: { rowData: any }) =>
+    this.props.selectOnRowClick && this.handleSelection(rowData)();
+
+  renderTableHeader(parentWidth: number) {
+    const {
+      editable,
+      instantEdit,
+      extraHeaderButtons,
+      rowHeight,
+      tableHeaderLabel,
+      tableHeaderHeight,
+    } = this.props;
+
+    const { editMode, selectedRows } = this.state;
+
+    const shouldRender = editable || extraHeaderButtons!.length > 0 || !!tableHeaderLabel;
+
+    return shouldRender ? (
+      <TableHeader
+        editable={editable}
+        editMode={editMode}
+        instantEdit={instantEdit}
+        // eslint-disable-next-line react/jsx-handler-names
```

Comment:
Shouldn't disable these. Should be `onEnableEditMode={this.handleEnableEditMode}`.

Only disable when passing props through, like `onEnableEditMode={onEnableEditMode}`